### PR TITLE
Handles OAuth, UpApi object, and unit tests 

### PR DESCRIPTION
**Pull request by ray-ray**
Updated the spec to reflect this PR: go/upapisdk
The README in this PR explains how to use the SDK and what's implemented so far.

### CODE REVIEW

File: upapi/endpoints.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+"""
+This module holds all the UP API routes
+"""
+DOMAIN = 'https://jawbone.com'
+AUTH_PATH = '%s/auth/oauth2' % DOMAIN
```
Comment (ornj) [inline upapi/endpoints.py]: use `.format()`

File: upapi/__init__.py [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+                'auto_refresh_kwargs': {'client_id': self.app_id, 'client_secret': self.app_secret},
+                'token_updater': self.app_token_saver}
+
+        self.oauth = requests_oauthlib.OAuth2Session(
+            client_id=self.app_id,
+            scope=self.app_scope,
+            redirect_uri=self._redirect_uri,
+            token=self.token,
+            **refresh_kwargs)
+        self.oauth.headers['User-Agent'] = '%s %s' % (USERAGENT, self.oauth.headers['User-Agent'])
```
Comment (ornj) [inline upapi/__init__.py]: `.format()`

File: upapi/endpoints.py (hunk 1 above)
Comment (ray-ray) [inline upapi/endpoints.py]: Done

File: upapi/__init__.py (hunk 2 above)
Comment (ray-ray) [inline upapi/__init__.py]: Done

**Pull request merged**